docs/examples/tests/data_transfer_objects/test_dto_model_interaction.py

```diff
@@ -11,13 +13,15 @@
 )
 
 
+@pytest.mark.xfail(reason="sqlalchemy plugin model map returns wrong instance for test")
```

Comment:
These are failing as the new pattern keeps an instance of the plugin for the life of the factory type.. the plugin caches based on model name, and as we have multiple tests with the same `Company` model, the plugin is returning the pydantic model that is cached from the first test that has that model in it.

Reply:
How about resetting the model cache?

Reply:
the issue is that the classes are generated at load/compile time, so by the time we get to them in the tests, its already done. 

For example, I could just change the pattern from:

```py
class DataclassFactory(Factory[DataclassT]):
    """DTO Factory for dataclass models."""

    plugin: ClassVar[DataclassPlugin] = DataclassPlugin()
```

to

```py
class DataclassFactory(Factory[DataclassT]):
    """DTO Factory for dataclass models."""

    plugin_type: ClassVar[DataclassPlugin] = DataclassPlugin
```
And give every dto class generated by the factory its own plugin instance. I think that would solve it.. and it would let users have models with the same name across different modules without hitting the same issue also, so prob makes sense.